contrib/go/src/python/pants/contrib/go/tasks/go_compile.py

```diff
@@ -58,7 +60,8 @@ def execute(self):
                                                    vt.target.import_path + '.a')
 
   def _go_install(self, target, gopath):
-    args = self.get_options().build_flags.split() + [target.import_path]
+    build_flags = re.sub(r'^"|"$', '', self.get_options().build_flags)
```

Comment:
If we're going to go down the shlex split path, I think it might be worth it to extract a helper for doing the post-processing on the arguments and write unit tests for it.

For example, I think that the starting and ending quotes may be taken care of by the option parser, so this may not be necessary.